storage-drivers/s3.ts

```diff
@@ -0,0 +1,51 @@
+import * as Minio from 'minio'
+import { z } from 'zod'
+
+import { defineStorageDriver } from '@/lib/storage-driver'
+
+export const s3Driver = defineStorageDriver({
+  envSchema: z.object({
+    S3_BUCKET: z.string().min(1),
+    S3_ENDPOINT: z.string().min(1).default('s3.amazonaws.com'),
```

Comment:
We thought about the defaults and came to conclusion that it would be more transparent if we don't use defaults for endpoint, use ssl and port. 
It is more readable to manually specify these values. Also it shouldn't make so difficult to use AWS, because we already provide a template in the docs.